test/metabase/upload_test.clj

```diff
@@ -554,12 +528,30 @@
 
 (defn load-from-csv-and-sync!
   "Creates a table from the csv file using `load-from-csv!` in the session schema, if there is one. Returns a Table instance."
-  [driver db-id table-name file]
+  [driver db table-name file]
   (let [table-name        (ddl.i/format-name driver/*driver* table-name)
         schema            (sql.tx/session-schema driver/*driver*)
         schema+table-name (#'upload/table-identifier {:schema schema :name table-name})]
-    (@#'upload/load-from-csv! driver db-id schema+table-name file)
-    (sync-upload-test-table! :database (mt/db) :table-name table-name :schema-name schema)))
+    (@#'upload/load-from-csv! driver db schema+table-name file)
+    (sync-upload-test-table! :database db :table-name table-name :schema-name schema)))
+
+(defn- columns-with-auto-pk [columns]
+ (cond-> columns
+   (driver/database-supports? driver/*driver* :upload-with-auto-pk (mt/db))
+   (#'upload/columns-with-auto-pk)))
+
+(defn- header-with-auto-pk [header]
+  (cond->> header
+    (driver/database-supports? driver/*driver* :upload-with-auto-pk (mt/db))
+    (cons @#'upload/auto-pk-column-name)))
+
+(defn- rows-with-auto-pk [rows]
+  (cond->> rows
+    (driver/database-supports? driver/*driver* :upload-with-auto-pk (mt/db))
+    (map-indexed (fn [i row] (cons (inc i) row)))))
+
+(defn- column-position [table column-name]
+  (t2/select-one-fn :database_position Field :%lower.name (u/lower-case-en column-name) :table_id (:id table)))
```

Comment:
This extra coupling down to the model is unfortunate, but I guess it already has a huge blast radius.

Reply:
Yeah, at least it's better than before though. It's in one place now rather than in every test